### PR TITLE
Initial Flat Structure

### PR DESCRIPTION
**Pull request by michalpristas**
Edited package to pull, untar and move beats to be in flat structure, this most likely will change a bit with inputs but not that much

This PR should not make shipper work but instead keep beats working correctly
cc @blakerouse  for spec definition

### CODE REVIEW

**Comment by ph** [comment]
@michalpristas is that ready for a review, you mentioned its a draft? I just want to understand the expectation.

**Comment by michalpristas** [comment]
@ph ready for review. i will rename as @blakerouse suggested but aside from that this is a first step.

File: pkg/component/component.go [hunk 1]
```diff
@@ -0,0 +1,93 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
```
Comment (michalpristas) [inline pkg/component/component.go:1]: this needs to be updated with component.go from blakes PR.
this has a lot of dumb logic and not implemented handlers which blake already solved so i will wait for him to push changes in

**Comment by michalpristas** [comment]
@blakerouse conflicts are fixed
i'm waiting for feedback from @simitt about beats in cloud image to see if i can remove them or figure out workarounds. why they are needed and so

File: dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl [hunk 2]
```diff
@@ -19,11 +19,7 @@ RUN mkdir -p {{ $beatHome }}/data {{ $beatHome }}/data/elastic-agent-{{ commit_s
     chmod 0755 {{ $beatHome }}/data/elastic-agent-*/elastic-agent && \
 {{- range $i, $modulesd := .ModulesDirs }}
     chmod 0775 {{ $beatHome}}/{{ $modulesd }} && \
-{{- end }}
```
Comment (michalpristas) [inline dev-tools/packaging/templates/docker/Dockerfile.elastic-agent.tmpl:22]: this beats mentioned in comment aboce

File: pkg/component/component_set.go [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package component
+
+import (
+	"github.com/pkg/errors"
+)
+
+type ComponentSet map[string]Spec
```
Comment (michalpristas) [inline pkg/component/component_set.go:11]: this will be removed as soon as we switch from beats

**Comment by michalpristas** [comment]
i modified build process, so far i'm not breaking cloud build

**Comment by simitt** [comment]
> i'm waiting for feedback from @simitt about beats in cloud image to see if i can remove them or figure out workarounds. why they are needed and so

Filebeat and metricbeat are bundled to collect logs and metrics when enabled. Please discuss any specificas directly with @elastic/cloud-observability or @mieciu.

**Comment by michalpristas** [comment]
not breaking cloud build, copying metric/filebeat as intended before the change
we can discuss about possibility of replacing those

**Review by blakerouse — APPROVED** [review]
Thanks for simplifying the PR, made it much easy to review and allows an easier transition to v2. I can use this to start executing the binaries from the components directory with the new v2 runtime.

**Pull request merged**